cloud_foundry_api/datadog_checks/cloud_foundry_api/cloud_foundry_api.py

```diff
@@ -0,0 +1,143 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import time
+from typing import Any, Dict
+
+from dateutil import parser
+from requests.exceptions import HTTPError
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.types import Event
+
+
+class CloudFoundryApiCheck(AgentCheck):
+    SOURCE_TYPE_NAME = "CloudFoundry"
+    __NAMESPACE__ = "cloud_foundry_api"
+
+    def __init__(self, name, init_config, instances):
+        super(CloudFoundryApiCheck, self).__init__(name, init_config, instances)
+
+        self._api_url = self.instance.get("api_url")
+        self._uaa_url = self.instance.get("uaa_url")
+        self._client_id = self.instance.get("client_id")
+        self._client_secret = self.instance.get("client_id")
+        self._api_version = self.instance.get("api_version", "v3")
+        self._event_filter = ",".join(self.instance.get("event_filter", []))
+        self._tags = self.instance.get("tags", [])
+        self._per_page = self.instance.get("results_per_page", 2000)
+        self._last_event_guid = ""
+        self._last_event_ts = 0
+        self._oauth_token = None
+        self._token_expiration = 0
+
+    def get_oauth_token(self):
+        # type: () -> str
+        if self._oauth_token is not None and self._token_expiration > int(time.time()):
```

Comment:
Should we add some buffer to the condition? Otherwise it may expire mid pagination in the get_events.